### ISSUE TITLE
Performance drop during training

### ISSUE THREAD

**Issue by njustesen** [Unity-Technologies/ml-agents#2897]
I am running into a problem quite often where the performance suddenly drops during training. I think this happens when the agent for the first time observes a reward that is much larger than previously observed - but not entirely sure. I give rewards in [0,1] as suggested in the best practices doc. Small rewards are given at each step in [0.0001 - 0.001] and a large reward when completing the level in [0-1].

![Screenshot 2019-11-12 at 17 41 02](https://user-images.githubusercontent.com/1433421/68691380-efc0f680-0573-11ea-974a-2effcf77fcfb.png)

![Screenshot 2019-11-12 at 17 49 34](https://user-images.githubusercontent.com/1433421/68691906-d40a2000-0574-11ea-89a2-5140d8f6caed.png)

My questions are:
- Doesn't PPO use gradient clipping to avoid this kind of drastic change? 
- Should I change my reward function or can I do something else?

I am using a custom environment with this configuration:
```
default:
    trainer: ppo
    epsilon: 0.2
    lambd: 0.95
    learning_rate: 3.0e-4
    learning_rate_schedule: linear
    memory_size: 256
    normalize: false
    sequence_length: 64
    summary_freq: 1000
    use_recurrent: false
    vis_encode_type: simple
    time_horizon: 128
    batch_size: 256
    buffer_size: 16384
    hidden_units: 512
    num_layers: 2
    beta: 1.0e-2
    max_steps: 1.0e7
    num_epoch: 3
    reward_signals:
        extrinsic:
            strength: 1.0
            gamma: 0.99
```

Some more details:
![Screenshot 2019-11-12 at 17 46 34](https://user-images.githubusercontent.com/1433421/68691674-6a8a1180-0574-11ea-9c81-c938f65af994.png)
